pkg/worker/request_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package worker
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+func TestErrorRequestRetry(t *testing.T) {
+
+	tests := []struct {
+		name string
+		f    func() (io.Reader, string, error)
+	}{
+		{
+			name: "error request retry",
+			f: func() (io.Reader, string, error) {
+				return nil, "", errors.New("didn't succeed")
+			},
+		},
+		{
+			name: "success request retry",
+			f: func() (io.Reader, string, error) {
+				return bytes.NewBuffer([]byte("success!")), "success!", nil
+			},
+		},
+	}
+
+	for _, test := range tests {
+
+		t.Run(test.name, func(t *testing.T) {
+			testServer := &testServer{
+				responseCodes: []int{500, 200},
+			}
+
+			server := httptest.NewTLSServer(testServer)
```

Comment:
Can this be moved outside of the for loop?

Reply:
no, because the test-server's state needs to be reinitialised for every loop.

Reply:
oh, I guess I can just set the handler